src/compositemark/callout.ts

```diff
@@ -1,30 +1,25 @@
 import {Config} from '../config';
 import {Encoding} from '../encoding';
 import * as log from '../log';
-import {MarkConfig, MarkDef} from '../mark';
+import {MarkConfig} from '../mark';
 import {GenericUnitSpec, LayerSpec} from '../spec';
 
-
 export const CALLOUT: 'callout' = 'callout';
 export type CALLOUT = typeof CALLOUT;
 
-export interface CalloutDef {
+export interface CalloutDef extends CalloutConfig {
   type: CALLOUT;
-  calloutAngle?: number;
-  calloutOffset?: number;
-  calloutLength?: number;
-  labelOffset?: number;
-  callout?: MarkDef;
-  label?: MarkDef;
+  callout?: MarkConfig;
+  label?: MarkConfig;
 }
 
 export function isCalloutDef(mark: CALLOUT | CalloutDef): mark is CalloutDef {
   return !!mark['type'];
 }
 
-export type CalloutStyle = 'callout-label' | 'callout-rule';
+export type CalloutStyle = 'calloutLabel' | 'calloutRule';
```

Comment:
Wonder why you prefer camelcase?   (I start liking dash more at this point, but wanna hear your rationale too.)

Reply:
My rationale is we most use camelCase for property keys, but dash-case for property values.

Since style name is a value, then dash-case is better.